notalib/date.py

```diff
@@ -41,27 +42,69 @@ class WeekNumbering(Enum):
 	MATCH_YEAR = auto()
 
 
-def parse_month(yyyy_mm):
+def parse_month(yyyy_mm: str) -> Tuple[int, int]:
+	"""
+	Parses string which contains month and year.
+	Expected format: 'YYYY-M'
+
+	Args:
+		yyyy_mm: Year and month to be parsed.
+
+	Returns:
+		Tuple where the first value is year and the second is month.
+
+	Raises:
+		ValueError: When data could not be extracted.
+	"""
 	a = arrow.get(yyyy_mm, 'YYYY-M')
-	return (a.year, a.month)
+	return a.year, a.month
+
+
+def parse_date(s: str, format_or_formats: DateFormats) -> arrow.Arrow:
+	"""
+	Tries to parse the date according to the specified format or formats.
+
+	Args:
+		s: Date to be parsed.
+		format_or_formats: Possible format or formats for date parsing.
 
+	Raises:
+		ValueError: When date can't be parsed.
 
-def parse_date(s, format_or_formats):
-	d = None
+	Returns:
+		arrow.Arrow object.
+	"""
 	for f in ensure_iterable(format_or_formats):
 		try:
-			d = arrow.get(s, f)
-			break
-		except:  # should probably narrow to arrow.ParserError
+			return arrow.get(s, f)
+
+		except ValueError:
```

Comment:
`arrow.ParseError` is subclass of `ValueError` and i use it because parsing wrong date ('2022-12-41') will raise `ValueError` instead of `arrow.ParseError`